tests/kubernetes-plugins/basic.bats

```diff
@@ -0,0 +1,83 @@
+    run kubectl delete namespace "$TEST_NAMESPACE"
+    run kubectl create namespace "$TEST_NAMESPACE"
+    # HELM_VALUES_EXTRA_FILE is a default file containing global helm
+    # options that can be optionally applied on helm install/upgrade
+    # by the test. This will fall back to $TEST_ROOT/defaults/values.yaml.tpl
+    # if not passed.
+    if [ -e  "${HELM_VALUES_EXTRA_FILE}" ]; then
+      envsubst < "${HELM_VALUES_EXTRA_FILE}" > "${HELM_VALUES_EXTRA_FILE%.*}"
+      export HELM_VALUES_EXTRA_FILE="${HELM_VALUES_EXTRA_FILE%.*}"
+    fi
+}
+
+teardown() {
+    if [[ "${SKIP_TEARDOWN:-no}" != "yes" ]]; then
+        run kubectl delete namespace "$TEST_NAMESPACE"
+        rm -f ${HELM_VALUES_EXTRA_FILE}
+    fi
+}
+
+# These are required for bats-detik
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAME="kubectl -n $TEST_NAMESPACE"
+# shellcheck disable=SC2034
+DETIK_CLIENT_NAMESPACE="${TEST_NAMESPACE}"
+
+
+@test "test fluent-bit reads k8s labels" {
+    helm repo add fluent https://fluent.github.io/helm-charts/ || helm repo add fluent https://fluent.github.io/helm-charts
+    helm repo update --fail-on-repo-update-fail
+
+    helm upgrade --install --debug --create-namespace --namespace "$TEST_NAMESPACE" fluent-bit fluent/fluent-bit \
+        --values ${BATS_TEST_DIRNAME}/resources/fluentbit-basic.yaml \
+        --set image.repository=${FLUENTBIT_IMAGE_REPOSITORY},image.tag=${FLUENTBIT_IMAGE_TAG},env[0].name=TEST_NAMESPACE,env[0].value=${TEST_NAMESPACE} \
+        --values "$HELM_VALUES_EXTRA_FILE" \
+        --timeout "${HELM_FB_TIMEOUT:-5m0s}" \
+        --wait
+
+    try "at most 15 times every 2s " \
+        "to find 1 pods named 'fluent-bit' " \
+        "with 'status' being 'running'"
+
+    # The hello-world-1 container MUST be on the same node as the fluentbit worker, so we use a nodeSelector to specify the same node name
+    run kubectl get pods -l "app.kubernetes.io/name=fluent-bit" -o jsonpath='{.items[0].spec.nodeName}'
+    assert_success
+    node_name=$output
+
+    run kubectl run -n $TEST_NAMESPACE hello-world-1 --image=docker.io/library/alpine:latest -l "this_is_a_test_label=true" \
+        --overrides="{\"apiVersion\":\"v1\",\"spec\":{\"nodeSelector\":{\"kubernetes.io/hostname\":\"$node_name\"}}}" \
+        --command -- sh -c 'while true; do echo "hello world"; sleep 1; done'
+
+    try "at most 15 times every 5s " \
+        "to find 1 pods named 'hello-world-1' " \
+        "with 'status' being 'Running'"
+
+    # We are sleeping here specifically for the Fluent-Bit's tail input's
+    # configured Refresh_Interval to have enough time to detect the new pod's log file
+    # and to have processed part of it
+    sleep 3
+
+    run kubectl logs -l "app.kubernetes.io/name=fluent-bit" -n "$TEST_NAMESPACE"
+    match1='kubernetes":{"pod_name":"hello-world-1","namespace_name":'
+    match1=${match1}\"${TEST_NAMESPACE}\"
+    match2='"labels":{"this_is_a_test_label":"true"}'
+
+    assert_output --partial $match1
```

Comment:
we do have a wait for output function actually we could reuse from our Calyptia one rather than having to rely on sleep always working: https://github.com/calyptia/bats/blob/main/test-helpers.bash#L121

We can probably copy over the test helpers to here to use it

Reply:
ah, ignore me - that's for local containers and not pods

Reply:
That looks like it's just for the container's log itself, we can get that pretty quickly (ie seeing the output of the fluent-bit startup/version text). This is a mildly different as the sleep is specifically to ensure that the input `tail` plugin has had enough time for the configured `Refresh_Interval` so that it can pick up the container log for the new hello-world-1 container and process it. We could look into a future improvement that adds a k8s service for the http service in fluent and then use a curl/etc to get the metrics that tells you the input plugin has processed records, but the `sleep` for now doesn't feel that bad to me 😄

Reply:
let's make the sleep plenty long then, say 10s just to ensure we definitely have it and prevent any flakes at the expense of test time

Reply:
done! just force pushed the last update